cmd/containerd-shim-runhcs-v1/serve.go

```diff
@@ -112,10 +130,14 @@ var serveCommand = cli.Command{
 				lerrs <- err
 				return
 			}
-			// Switch the logrus output to here. Note: we wont get this
```

Comment:
Sound we even run this goroutine at all for anything but NPIPE? It doesnt really make sense to have the background listener and connection if we aren't going to use it right? Just a waste of resources.

Reply:
I thought containerd might treat it as an error if it never gets a connection on the pipe, but I can try to validate that assumption.

Reply:
We should verify. It doesnt for the 2nd and nth shims. We should make sure it doesnt though because in this mode we really dont want the connection.

Reply:
From the containerd code, it looks like we won't log an error if we fail to connect to the log pipe due to file not found (this is from your change to prevent the "copy shim log" error). So it should work to not create the log pipe at all unless we need it. To confirm I also tested with log set to ETW and didn't see any errors from containerd.